src/main/java/core/basesyntax/strategy/impl/FruitShopTransactionImpl.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.strategy.BalanceStrategy;
+import core.basesyntax.strategy.FruitShopTransaction;
+import core.basesyntax.strategy.PurchaseStrategy;
+import core.basesyntax.strategy.ReturnStrategy;
+import core.basesyntax.strategy.SupplyStrategy;
+import java.util.Map;
+
+public class FruitShopTransactionImpl implements FruitShopTransaction {
+    private static final String BALANCE = "b";
+    private static final String SUPPLY = "s";
+    private static final String PURCHASE = "p";
+    private static final String RETURN = "r";
+    private final BalanceStrategy balanceStrategy;
```

Comment:
Better will be use - new BalanceStrategyImpl().action(dataForReport, fruit, quantity); 
Than you don't 15 - 25 lines